automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/SamApiSpec.scala

```diff
@@ -0,0 +1,33 @@
+package org.broadinstitute.dsde.firecloud.test.api
+
+import org.broadinstitute.dsde.firecloud.api.Sam
+import org.broadinstitute.dsde.firecloud.api.Sam.user.UserStatus
+import org.broadinstitute.dsde.firecloud.auth.{AuthToken, ServiceAccountAuthToken, UserAuthToken}
+import org.broadinstitute.dsde.firecloud.config.{Credentials, UserPool}
+import org.scalatest.{FreeSpec, Matchers}
+
+class SamApiSpec extends FreeSpec with Matchers {
+  val anyUser: Credentials = UserPool.chooseAnyUser
+  val userAuthToken: AuthToken = UserAuthToken(anyUser)
+
+  "Sam" - {
+    "should give pets the same access as their owners" in {
+
+      // set auth tokens explicitly to control which credentials are used
+
+      val userInfo: UserStatus = Sam.user.status()(userAuthToken)
+
+      val petAccountEmail = Sam.user.petServiceAccountEmail()(userAuthToken)
+
+      // first call should create pet.  confirm that a second call to create/retrieve gives the same results
```

Comment:
Does this mean that the conditions of the test are different the second time it's run? For a FiaB instance? Forever?

Reply:
The Sam API does a "get or create". So the first one creates the pet in Google and in LDAP. The second call should return the created pet, not create a new one.

LDAP is local to the fiab, but are the pets removed in Google on test tear down?

Reply:
I'll add a pet cleaning service

Reply:
Based on my experience with writing a registration test, I recommend cleaning your pet before the test as well as after just in case it wasn't cleaned up correctly for some reason. Also, can you fail the test if the pet isn't clean? Otherwise the test could succeed but not really be testing what it claims to.

Reply:
Here's the plan:
* ask Google if the pet exists -> fail if true
* create pet
* ask Google if the Pet exists -> fail if false
* [more tests]
* remove pet
* ask Google if the pet exists -> fail if true

Reply:
Sounds reasonable. You should be able to use `GoogleIamDAO. findServiceAccount/removeServiceAccount` for this.